### PR TITLE
Adding to README of tinystr/TinyAsciiStr

### PR DESCRIPTION
**Pull request by andrewpollack**
While working on `tinystr` and `TinyAsciiStr`, I've added a small blurb onto the README with what I've found

### CODE REVIEW

**Comment by zbraniecki** [comment]
Thank you for writing this! 
Can you add a bit on when TinyStr is the right choice? 
You can see some of this in the old readme - https://github.com/zbraniecki/tinystr

Basically - whenever you need to compare strings, lowercase/uppercase/titlecase, TinyStr is the edge performance library.

**Comment by andrewpollack** [comment]
> Thank you for writing this! Can you add a bit on when TinyStr is the right choice? You can see some of this in the old readme - https://github.com/zbraniecki/tinystr
> 
> Basically - whenever you need to compare strings, lowercase/uppercase/titlecase, TinyStr is the edge performance library.

Thank you for the suggestion and reference! Would I have permission to pull from the examples in that README?

File: utils/tinystr/src/lib.rs [hunk 1]
```diff
@@ -2,6 +2,51 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+//! `tinystr` is a utility crate of the [`ICU4X`] project.
+//!
+//! It includes [`TinyAsciiStr`], a core API for representing small ASCII-only bounded length strings.
+//!
+//! It is optimized for operations on strings of size 8 or smaller. When use cases involve comparison and conversion of strings for lowercase/uppercase/titlecase, or checking numeric/alphabetic/alphanumeric, `TinyAsciiStr` is the edge performance library.
```
Comment (Manishearth) [inline utils/tinystr/src/lib.rs:9]: nit: i would wrap this text to fit at around 100 chars (no need to be super exact)

Comment (andrewpollack) [reply utils/tinystr/src/lib.rs]: Good call -- updated

**Review by zbraniecki — APPROVED** [review]
lgtm!

**Comment by Manishearth** [comment]
Thanks!

**Pull request merged**